extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsProducer.java

```diff
@@ -103,7 +105,9 @@ public KafkaStreamsTopologyManager kafkaStreamsTopologyManager() {
     }
 
     void onStop(@Observes ShutdownEvent event) {
-        shutdown = true;
+        if (!ignoreShutdownFlag) {
```

Comment:
It may be simpler to set _shutdown_ to false in KafkaStreamsProducer constructor ?
If injection is being re-triggered, I suppose it means a restart is happening in same classloader.

Reply:
makes sense, tested it works